### DISCUSSION TITLE
Unable to Get complete Email content

### DISCUSSION THREAD

**Discussion by sabeehsharif** [microsoft/hack-together#65] [Q&A]
Hi, 

**Topic: Email Content**

I can get email body but can't get the full content? using this function await _graphServiceClient.Me.MailFolders.Inbox.Messages 
these all messages are coming but content is not complete of individual email body. 

anyone can assist me? @waldekmastykarz

**Comment by aycabas** [comment]
Hi @sabeehsharif, you'll need to define a header `Prefer: outlook.body-content-type="text"` to see the body in email message. Here is how you can get email body in text format: 
```
var graphClient = new GraphServiceClient(requestAdapter);

var result = await graphClient.Me.Messages["{message-id}"].GetAsync((requestConfiguration) =>
{
	requestConfiguration.QueryParameters.Select = new string []{ "subject","body","bodyPreview","uniqueBody" };
	requestConfiguration.Headers.Add("Prefer", "outlook.body-content-type=\"text\"");
});
```

Please check out the documentation for more details and let us know if you need further assistance: https://learn.microsoft.com/en-us/graph/api/message-get?view=graph-rest-1.0&tabs=csharp#request-2

**Reply by sabeehsharif** [reply]
Thanks @aycabas  i implemented but still getting half content of email body.

I think Overall header option is not working, i am doing this to get complete text of email body but after using these headers still getting half content of email body.

`List<Option> options = new List<Option> { //Creating header new HeaderOption("Prefer", "outlook.body-content-type='text'") }; var emails = await _graphServiceClient.Me.Messages[id].Request(options).Select(msg => new { msg.Subject, msg.BodyPreview, msg.ReceivedDateTime, msg.Id, msg.InternetMessageId, msg.ConversationId }).GetAsync();
`


List options = new List
{
//Creating header
new HeaderOption("Prefer", "outlook.body-content-type='text'")
};
var emails = await _graphServiceClient.Me.Messages[id].Request(options).Select(msg => new
{
msg.Subject,
msg.BodyPreview,
msg.ReceivedDateTime,
msg.Id,
msg.InternetMessageId,
msg.ConversationId
}).GetAsync();

**Reply by aycabas** [reply]
Hi @sabeehsharif, I realize that you are calling "BodyPreview", not the body. The entire message content is available under the Body > Content, BodyPreview only brings you the first couple words from the mail. Please revisit the example I shared earlier, you need to capture content >> inside body parameter, not the BodyPreview: https://learn.microsoft.com/en-us/graph/api/message-get?view=graph-rest-1.0&tabs=http#request-2

Let me know how it goes :)

**Reply by sabeehsharif** [reply]
Thank you so much @aycabas working perfectly now.
